atomic_reactor/plugins/add_filesystem.py

```diff
@@ -344,10 +344,13 @@ def _add_filesystem_to_dockerfile(self, file_name, build_dir: BuildDir):
 
         build_dir.dockerfile.lines = lines
         new_parents = []
+        df_images = self.workflow.data.dockerfile_images
 
         for image in build_dir.dockerfile.parent_images:
             if base_image_is_custom(image):
                 new_parents.append('scratch')
+                # To make change_from_in_df happy and nothing will change due to this.
+                df_images[image] = 'scratch'
```

Comment:
Or maybe, instead of this workaround, could we update `change_from_in_df` to skip the unresolved check for custom base images? I don't really know if that's any better though...

Reply:
@chmeliik Just see your comment here. After testing a few times, a new solution is introduced linked below. The change in this PR introduces subsequent new issues :disappointed: